specs/AppLifecycle/Instancing/AppLifecycle SingleMulti-Instancing.md

```diff
@@ -0,0 +1,422 @@
+        // activation kinds.
+        DoSomethingWithNewActivationArgs(args);
+    }
+}
+```
+
+In the existing UWP API, RedirectActivationTo does not take any parameters. However, in the UWP
+case, the activation contract is tightly controlled, and we could ensure that the arguments passed
+on to the app were the arguments for the redirected instance. In bringing this feature to Win32, we
+can't control activation as tightly. Also, the ActivationArguments object returned by
+GetActivatedEventArgs is not the same object as the IActivatedEventArgs object that is passed in to
+the Activated event callback. The new API gives the app the opportunity to modify or replace the
+payload to be passed to the target instance.
+
+### Unregister for redirection
+
+Just as an app can define its own logic for how and when it registers an instance for redirection,
+so too will it be able to define its own logic for when a given instance no longer wants to be a
+target for redirection.
+
+Following the previous example of an app that supports filetype activation, and redirects to the
+instance that already has the selected file open - when that file is closed, the app could
+unregister itself.
+
+```c++
+void CALLBACK OnFileClosed(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    AppInstance::GetCurrent().UnregisterKey(g_fileName);
+}
+```
+
+### Instance information
+
+The AppInstance class is intended to represent an instance of the app. In the initial release, the
+API is focused on instance redirection, and therefore exposes methods and properties relevant to
+that operation. However, the intention is that the class will expand in later releases to include
+other methods and properties that would be relevant for an app instance. In the first release, we
+also include the process ID, which the app may use for any purpose it sees fit, including telemetry
+or debugging.
+
+```c++
+void DumpExistingInstances()
+{
+    for (AppInstance const& instance : AppInstance::GetInstances())
+    {
+        std::wostringstream sStream;
+        sStream << L"Instance: ProcessId = " << instance.ProcessId
+            << L", Key = " << instance.Key().c_str() << std::endl;
+        ::OutputDebugString(sStream.str().c_str());
+    }
+}
+```
+
+## API Details
+
+This API extends the [AppLifecycle AppInstance](../Activation/AppLifecycle%20Activation.md) type
+with some additional methods for activation redirection. Specifically, `AppInstance` is expanded
+to include:
+
+```idl
+namespace Microsoft.Windows.AppLifecycle
+{
+    runtimeclass AppInstance
+    {
+        // Existing activation functionality like GetCurrent and GetActivatedEventArgs
+
+        static event EventHandler<ActivationArguments> Activated;
+
+        static Windows.Foundation.IVector<AppInstance> GetInstances();
+
+        static AppInstance FindOrRegisterForKey(String key);
+        void UnregisterKey(String key);
+
+        void RedirectTo(ActivationArguments args);
```

Comment:
I think the method name `RedirectActivationTo` is clearer about what is being redirected, especially since AppInstance is now being used for lots more than just activation redirection.

Reply:
RECOMMEND: Use `RedirectActivationTo`

Reply:
Fixed.